jba/src/processing/tasktracker_content_collector.py

```diff
@@ -0,0 +1,128 @@
+)
+
+CONTENT_FILE_NAME = 'task_content_default.yaml'
+FRAMEWORK_TYPE = 'framework'
+TASK_DIRECTORY_NAME = 'task'
+EXTENSIONS = {'py': 'PYTHON', 'ipynb': 'JUPYTER', 'java': 'JAVA', 'kt': 'KOTLIN', 'cpp': 'CPP', 'csv': 'CSV'}
+
+
+@dataclass(eq=True, frozen=True)
+class TaskTrackerFile:
+    rel_path: Path
+
+    @property
+    def name(self) -> str:
+        return self.rel_path.stem
+
+    @property
+    def path(self) -> Path:
+        return self.rel_path.parent
+
+    @property
+    def extension(self) -> str:
+        return EXTENSIONS.get(self.rel_path.suffix.lstrip('.'), 'NO_EXTENSION')
+
+    def as_dict(self) -> dict:
+        return {
+            'filename': self.name,
+            'relativePath': str(self.path),
+            'extension': self.extension,
+            'sourceSet': 'SRC',
+            'isInternal': False
+        }
+
+
+def get_data_template(files: List[Dict]) -> Dict:
+    return {
+        'tasks': [{
+            'name': 'example',
+            'description': 'description',
+            'id': 'main',
+            'files': files
+        }]
+    }
+
+
+def get_info_file(root: Path) -> Path:
+    info_files = find_files_by_regex(root, INFO_FILE_REGEX)
+    return info_files[0]
+
+
+def get_files(course_root: Path, relative_path: Path, lesson: EduStructureNode) -> List[TaskTrackerFile]:
+    lesson_path = course_root / relative_path
+    info_file = get_info_file(lesson_path)
+    yaml_file_content = read_yaml_field_content(lesson_path / info_file, EduInfoFileField.TYPE.value)
+    return itertools.chain.from_iterable(
+        [get_task_files(lesson_path / child.name,
+                        relative_path,
+                        yaml_file_content is not None and yaml_file_content == FRAMEWORK_TYPE)
+         for child in lesson.children])
+
+
+def get_task_files(root: Path, relative_path: Path, is_framework: bool) -> List[TaskTrackerFile]:
+    info_file = get_info_file(root)
+
+    files = read_yaml_field_content(root / info_file, EduInfoFileField.FILES.value)
+    if files is None:
+        files = []
+    files = list(filter(lambda file: file[EduInfoFileField.VISIBLE.value], files))
+
+    def get_filename(file_content: Dict) -> TaskTrackerFile:
+        if is_framework:
+            return TaskTrackerFile(relative_path / TASK_DIRECTORY_NAME / file_content[EduInfoFileField.NAME.value])
+        return TaskTrackerFile(relative_path / root.name / file_content[EduInfoFileField.NAME.value])
+
+    return list(map(get_filename, files))
+
+
+def get_yaml_content(course_root: Path) -> Dict:
+    course_structure = gather_structure(course_root)
+    lessons = course_structure.gather_leafs_of_type(EduStructureType.LESSON)
+    files = {
+        file
+        for path, lesson in lessons.items()
+        for file in get_files(course_root, Path(*path[1:]), lesson)
+    }
+    return get_data_template(list(map(lambda obj: obj.as_dict(), files)))
+
+
+def configure_parser(parser: argparse.ArgumentParser) -> None:
```

Comment:
Let's swap `destination_path` and `course_sources_path`, because it's strange that you have to enter output first and then enter input